mytoyota/models/endpoints/v1_location.py

```diff
@@ -0,0 +1,31 @@
+""" Toyota Connected Services API - V1 Location Models """
+from datetime import datetime
+from typing import Optional, List, Any, Union
+from pydantic import BaseModel, Field
+
+from .common import _StatusModel
+
+# pylint: disable=locally-disabled, missing-class-docstring, fixme
+
+
+class _VehicleLocationModel(BaseModel):
+    display_name: str = Field(alias="displayName")
+    latitude: float
+    location_acquisition_datetime: datetime = Field(alias="locationAcquisitionDatetime")
+    longitude: float
+
+
+class _LocationModel(BaseModel):
+    last_timestamp: Optional[datetime] = Field(alias="lastTimestamp", default=None)
+    vehicle_location: Optional[_VehicleLocationModel] = Field(
+        alias="vehicleLocation", default=None
+    )
+    vin: Optional[str] = None
+
+
+class V1LocationModel(BaseModel):
```

Comment:
If we plan to output the status in each endpoint model, we can also simply take the `_StatusModel` as the basis of the respective endpoint model and inherit from it.

Reply:
Yes I had the same idea but if you look at some of the older V1 models, they return different data in the status field depending on if the call was successful or not. If you take a look at `v1_location_ok.json` and `v1_location_error.json' in the tests you can see the two different responses. I suppose we could model this though?

Reply:
That does indeed look odd 😆 
In this case, it should be possible to model this via a union:

```python
""" Toyota Connected Services API - Commend Endpoint Models """
from typing import List, Optional, Union
from pydantic import BaseModel, Field


class _MessageModel(BaseModel):
    description: str
    detailed_description: Optional[str] = Field(
        alias="detailedDescription", default=None
    )
    response_code: str = Field(alias="responseCode")


class _MessagesModel(BaseModel):
    messages: List[_MessageModel]


class _StatusModel(BaseModel):
    status: Union[str, _MessagesModel]
    code: Optional[int] = None
    errors: Optional[List] = []
    message: Optional[str] = None
```